### ISSUE TITLE
Corrected.bw and uncorrected.bw identical aggregate signal?

### ISSUE THREAD

**Issue by vkartha** [loosolab/TOBIAS#59]
Hi,
    Is there any reason we won't get any negative values when visualizing the aggregate profile using the corrected.bw files (visualizing across all sites)? I realized this when trying to run the CoverageHeatmap function, and the color scales default to blue red (-/+), and I was only seeing white and red color mappings. I'm interested in comparing two conditions, and the differences I'm picking up make sense, but wanted to make sure the normalizations / scales aren't off for any reason when visualizing the footprints across the two conditions. I followed the default ATACorrect run to generate the resulting .bw files as per the manual. 
    
 Thanks again!

**Comment by vkartha** [comment]
On closer inspection, it appears that comparing the uncorrected.bw to the corrected.bw for a set of binding sites suggests that they are in fact identical. Is there a reason this can happen? Pasting image of aggregate profile with corrected signal on the left and uncorrected on the right

![image](https://user-images.githubusercontent.com/9615730/107677532-5eee3480-6c68-11eb-92bd-4f5894fd1075.png)

**Comment by vkartha** [comment]
Sorry for the troubles, quick follow up! I think the issue is related to my bam file input. Originally I was working with fragments files (similar to bed format, with each entry being a fragment from paired-end reads, the start and end of each fragment representing the Tn5 cut sites). Here's  an example of such a fragment file (the last column is just a grouping variable for the sample the fragment comes from):

chr1	713944	713998	Donor2_S1_BC065_N01
chr1	713945	713998	Donor2_S1_BC065_N01
chr1	713957	714004	Donor2_S1_BC356_N01
chr1	713972	714058	Donor2_S1_BC041_N01
chr1	713972	714120	Donor2_S1_BC041_N01
chr1	713973	714049	Donor2_S1_BC065_N01
chr1	713982	714170	Donor2_S1_BC053_N01
chr1	714040	714134	Donor2_S1_BC053_N01
chr1	714064	714197	Donor2_S1_BC312_N01
chr1	714119	714212	Donor2_S1_BC041_N01

I then re-format this file into a .bam file using bedpeToBam from bedtools (such that I keep the start and end of each fragment as entries to indicate cut-sites from paired-end reads, assuming that this is what TOBIAS expects). Here is an example of such a resulting bam file (generated for the above fragments):
 
Fragment1	99	chr1	713944	255	1M	=	713998	54	*	*
Fragment1	147	chr1	713998	255	1M	=	713944	54	*	*
Fragment2	99	chr1	713945	255	1M	=	713998	53	*	*
Fragment2	147	chr1	713998	255	1M	=	713945	53	*	*
Fragment3	99	chr1	713957	255	1M	=	714004	47	*	*
Fragment3	147	chr1	714004	255	1M	=	713957	47	*	*
Fragment4	99	chr1	713972	255	1M	=	714058	86	*	*
Fragment4	147	chr1	714058	255	1M	=	713972	86	*	*
Fragment5	99	chr1	713972	255	1M	=	714120	148	*	*
Fragment5	147	chr1	714120	255	1M	=	713972	148	*	*
.........
........

I then use this bam file as input to TOBIAS ATACorrect, with default settings. 

So ultimately my question becomes whether giving such a bam (that just includes the Tn5 cut sites as single bp entries per read mate), is problematic and resulting in this issue (where the corrected and uncorrected .bw files are the same)? Is TOBIAS always expecting paired-end read entries vs single-end reads? This would dictate how I reformat the bam file to work correctly with TOBIAS.

**Comment by msbentsen** [comment]
Hi,

TOBIAS can work with both paired- and single-end reads, so this should not be the issue. But I am wondering what is going wrong with the `corrected == uncorrected` issue. Could you share the log-file with me? It is the stdout output of the `ATACorrect` run starting with:
 ```# TOBIAS 0.12.7 ATACorrect (run started (.....) ```
 
 That would help me to see if there was any correction of the Tn5 bias!

**Comment by vkartha** [comment]
Thanks for your prompt response to my question! Pasting the log for ATACorrect run below:

```
# TOBIAS 0.12.0 ATACorrect (run started 2021-02-12 14:14:27.927386)
# Working directory: /mnt/home/footprint
# Command line call: TOBIAS ATACorrect --bam ./myBam.bam --genome /mnt/genomes/fasta/hg19/hg19.fa --peaks ../peakCalls/all_merged.fixedwidthpeaks.bed --outdir ./test --cores 4

# ----- Input parameters -----
# bam:  ./myBam.bam
# genome:       /mnt/genomes/fasta/hg19/hg19.fa
# peaks:        ../peakCalls/all_merged.fixedwidthpeaks.bed
# regions_in:   None
# regions_out:  None
# blacklist:    None
# extend:       100
# split_strands:        False
# norm_off:     False
# track_off:    []
# k_flank:      12
# read_shift:   [4, -5]
# bg_shift:     100
# window:       100
# score_mat:    DWM
# prefix:       myBam
# outdir:       /mnt/home/footprint/test
# cores:        4
# split:        100
# verbosity:    3


# ----- Output files -----
# /mnt/home/footprint/test/myBam_uncorrected.bw
# /mnt/home/footprint/test/myBam_bias.bw
# /mnt/home/footprint/test/myBam_expected.bw
# /mnt/home/footprint/test/myBam_corrected.bw
# /mnt/home/footprint/test/myBam_atacorrect.pdf


2021-02-12 14:14:27 (14499) [INFO]      ----- Processing input data -----
2021-02-12 14:14:27 (14499) [INFO]      Reading info from .bam file
2021-02-12 14:14:28 (14499) [INFO]      Reading info from .fasta file
2021-02-12 14:14:28 (14499) [INFO]      Processing input/output regions
2021-02-12 14:14:43 (14499) [STATS]     genome: 92 regions | 3137144693 bp | 100.00% coverage
2021-02-12 14:14:43 (14499) [STATS]     input_regions: 253795 regions | 3071403293 bp | 97.90% coverage
2021-02-12 14:14:43 (14499) [STATS]     output_regions: 219138 regions | 109569000 bp | 3.49% coverage
2021-02-12 14:14:43 (14499) [STATS]     peak_regions: 219138 regions | 65741400 bp | 2.10% coverage
2021-02-12 14:14:43 (14499) [STATS]     nonpeak_regions: 253795 regions | 3071403293 bp | 97.90% coverage
2021-02-12 14:14:43 (14499) [STATS]     blacklist_regions: 0 regions | 0 bp | 0.00% coverage

2021-02-12 14:14:43 (14499) [INFO]      ----- Estimating normalization factors -----
2021-02-12 14:14:43 (14499) [INFO]      Counting reads in peak regions
2021-02-12 14:14:43 (14499) [INFO]      Progress: 0%
2021-02-12 14:14:45 (14499) [INFO]      Progress: 4%
2021-02-12 14:14:45 (14499) [INFO]      Progress: 8%
2021-02-12 14:14:46 (14499) [INFO]      Progress: 12%
2021-02-12 14:14:46 (14499) [INFO]      Progress: 16%
2021-02-12 14:14:47 (14499) [INFO]      Progress: 18%
2021-02-12 14:14:47 (14499) [INFO]      Progress: 21%
2021-02-12 14:14:48 (14499) [INFO]      Progress: 24%
2021-02-12 14:14:48 (14499) [INFO]      Progress: 28%
2021-02-12 14:14:49 (14499) [INFO]      Progress: 32%
2021-02-12 14:14:49 (14499) [INFO]      Progress: 36%
2021-02-12 14:14:50 (14499) [INFO]      Progress: 39%
2021-02-12 14:14:50 (14499) [INFO]      Progress: 42%
2021-02-12 14:14:51 (14499) [INFO]      Progress: 45%
2021-02-12 14:14:51 (14499) [INFO]      Progress: 49%
2021-02-12 14:14:52 (14499) [INFO]      Progress: 53%
2021-02-12 14:14:52 (14499) [INFO]      Progress: 57%
2021-02-12 14:14:53 (14499) [INFO]      Progress: 61%
2021-02-12 14:14:53 (14499) [INFO]      Progress: 65%
2021-02-12 14:14:54 (14499) [INFO]      Progress: 69%
2021-02-12 14:14:54 (14499) [INFO]      Progress: 73%
2021-02-12 14:14:55 (14499) [INFO]      Progress: 77%
2021-02-12 14:14:55 (14499) [INFO]      Progress: 81%
2021-02-12 14:14:56 (14499) [INFO]      Progress: 85%
2021-02-12 14:14:56 (14499) [INFO]      Progress: 88%
2021-02-12 14:14:57 (14499) [INFO]      Progress: 92%
2021-02-12 14:14:57 (14499) [INFO]      Progress: 95%
2021-02-12 14:14:58 (14499) [INFO]      Progress: 100%

2021-02-12 14:14:58 (14499) [INFO]      Counting reads in nonpeak regions
2021-02-12 14:14:58 (14499) [INFO]      Progress: 0%
2021-02-12 14:15:00 (14499) [INFO]      Progress: 4%
2021-02-12 14:15:00 (14499) [INFO]      Progress: 8%
2021-02-12 14:15:01 (14499) [INFO]      Progress: 12%
2021-02-12 14:15:02 (14499) [INFO]      Progress: 16%
2021-02-12 14:15:02 (14499) [INFO]      Progress: 20%
2021-02-12 14:15:03 (14499) [INFO]      Progress: 24%
2021-02-12 14:15:04 (14499) [INFO]      Progress: 28%
2021-02-12 14:15:04 (14499) [INFO]      Progress: 32%
2021-02-12 14:15:05 (14499) [INFO]      Progress: 36%
2021-02-12 14:15:05 (14499) [INFO]      Progress: 39%
2021-02-12 14:15:06 (14499) [INFO]      Progress: 43%
2021-02-12 14:15:06 (14499) [INFO]      Progress: 46%
2021-02-12 14:15:07 (14499) [INFO]      Progress: 48%
2021-02-12 14:15:07 (14499) [INFO]      Progress: 52%
2021-02-12 14:15:08 (14499) [INFO]      Progress: 55%
2021-02-12 14:15:08 (14499) [INFO]      Progress: 59%
2021-02-12 14:15:09 (14499) [INFO]      Progress: 61%
2021-02-12 14:15:09 (14499) [INFO]      Progress: 64%
2021-02-12 14:15:10 (14499) [INFO]      Progress: 67%
2021-02-12 14:15:10 (14499) [INFO]      Progress: 71%
2021-02-12 14:15:11 (14499) [INFO]      Progress: 74%
2021-02-12 14:15:11 (14499) [INFO]      Progress: 76%
2021-02-12 14:15:12 (14499) [INFO]      Progress: 79%
2021-02-12 14:15:12 (14499) [INFO]      Progress: 83%
2021-02-12 14:15:13 (14499) [INFO]      Progress: 86%
2021-02-12 14:15:13 (14499) [INFO]      Progress: 89%
2021-02-12 14:15:14 (14499) [INFO]      Progress: 92%
2021-02-12 14:15:14 (14499) [INFO]      Progress: 96%
2021-02-12 14:15:15 (14499) [INFO]      Progress: 100%
2021-02-12 14:15:15 (14499) [STATS]     TOTAL_READS     745748
2021-02-12 14:15:15 (14499) [STATS]     PEAK_READS      354118
2021-02-12 14:15:15 (14499) [STATS]     NONPEAK_READS   391630
2021-02-12 14:15:15 (14499) [STATS]     LIB_NORM        13.40936
2021-02-12 14:15:15 (14499) [STATS]     FRiP    0.47485
2021-02-12 14:15:15 (14499) [STATS]     CORRECTION_FACTOR:      28.23917

2021-02-12 14:15:15 (14499) [INFO]      Started estimation of sequence bias...
2021-02-12 14:15:15 (14499) [INFO]      Progress: 0%
2021-02-12 14:15:17 (14499) [INFO]      Progress: 3%
2021-02-12 14:15:17 (14499) [INFO]      Progress: 4%
2021-02-12 14:15:18 (14499) [INFO]      Progress: 5%
2021-02-12 14:15:20 (14499) [INFO]      Progress: 6%
2021-02-12 14:15:20 (14499) [INFO]      Progress: 7%
2021-02-12 14:15:21 (14499) [INFO]      Progress: 8%
2021-02-12 14:15:22 (14499) [INFO]      Progress: 9%
2021-02-12 14:15:23 (14499) [INFO]      Progress: 10%
2021-02-12 14:15:24 (14499) [INFO]      Progress: 11%
2021-02-12 14:15:25 (14499) [INFO]      Progress: 14%
2021-02-12 14:15:26 (14499) [INFO]      Progress: 15%
2021-02-12 14:15:26 (14499) [INFO]      Progress: 17%
2021-02-12 14:15:27 (14499) [INFO]      Progress: 19%
2021-02-12 14:15:28 (14499) [INFO]      Progress: 21%
2021-02-12 14:15:28 (14499) [INFO]      Progress: 22%
2021-02-12 14:15:29 (14499) [INFO]      Progress: 24%
2021-02-12 14:15:29 (14499) [INFO]      Progress: 25%
2021-02-12 14:15:30 (14499) [INFO]      Progress: 26%
2021-02-12 14:15:30 (14499) [INFO]      Progress: 29%
2021-02-12 14:15:31 (14499) [INFO]      Progress: 30%
2021-02-12 14:15:31 (14499) [INFO]      Progress: 31%
2021-02-12 14:15:32 (14499) [INFO]      Progress: 34%
2021-02-12 14:15:33 (14499) [INFO]      Progress: 36%
2021-02-12 14:15:33 (14499) [INFO]      Progress: 37%
2021-02-12 14:15:34 (14499) [INFO]      Progress: 39%
2021-02-12 14:15:35 (14499) [INFO]      Progress: 40%
2021-02-12 14:15:36 (14499) [INFO]      Progress: 41%
2021-02-12 14:15:37 (14499) [INFO]      Progress: 43%
2021-02-12 14:15:38 (14499) [INFO]      Progress: 44%
2021-02-12 14:15:38 (14499) [INFO]      Progress: 45%
2021-02-12 14:15:40 (14499) [INFO]      Progress: 48%
2021-02-12 14:15:40 (14499) [INFO]      Progress: 49%
2021-02-12 14:15:41 (14499) [INFO]      Progress: 51%
2021-02-12 14:15:42 (14499) [INFO]      Progress: 53%
2021-02-12 14:15:42 (14499) [INFO]      Progress: 54%
2021-02-12 14:15:43 (14499) [INFO]      Progress: 57%
2021-02-12 14:15:44 (14499) [INFO]      Progress: 58%
2021-02-12 14:15:44 (14499) [INFO]      Progress: 59%
2021-02-12 14:15:45 (14499) [INFO]      Progress: 60%
2021-02-12 14:15:45 (14499) [INFO]      Progress: 61%
2021-02-12 14:15:47 (14499) [INFO]      Progress: 62%
2021-02-12 14:15:47 (14499) [INFO]      Progress: 63%
2021-02-12 14:15:48 (14499) [INFO]      Progress: 64%
2021-02-12 14:15:50 (14499) [INFO]      Progress: 65%
2021-02-12 14:15:51 (14499) [INFO]      Progress: 67%
2021-02-12 14:15:51 (14499) [INFO]      Progress: 68%
2021-02-12 14:15:52 (14499) [INFO]      Progress: 69%
2021-02-12 14:15:53 (14499) [INFO]      Progress: 71%
2021-02-12 14:15:54 (14499) [INFO]      Progress: 72%
2021-02-12 14:15:54 (14499) [INFO]      Progress: 74%
2021-02-12 14:15:56 (14499) [INFO]      Progress: 75%
2021-02-12 14:15:56 (14499) [INFO]      Progress: 76%
2021-02-12 14:15:57 (14499) [INFO]      Progress: 77%
2021-02-12 14:15:58 (14499) [INFO]      Progress: 79%
2021-02-12 14:16:00 (14499) [INFO]      Progress: 80%
2021-02-12 14:16:01 (14499) [INFO]      Progress: 81%
2021-02-12 14:16:01 (14499) [INFO]      Progress: 82%
2021-02-12 14:16:02 (14499) [INFO]      Progress: 83%
2021-02-12 14:16:02 (14499) [INFO]      Progress: 84%
2021-02-12 14:16:03 (14499) [INFO]      Progress: 86%
2021-02-12 14:16:03 (14499) [INFO]      Progress: 87%
2021-02-12 14:16:04 (14499) [INFO]      Progress: 88%
2021-02-12 14:16:04 (14499) [INFO]      Progress: 90%
2021-02-12 14:16:05 (14499) [INFO]      Progress: 92%
2021-02-12 14:16:06 (14499) [INFO]      Progress: 94%
2021-02-12 14:16:07 (14499) [INFO]      Progress: 97%
2021-02-12 14:16:08 (14499) [INFO]      Progress: 99%
2021-02-12 14:16:09 (14499) [INFO]      Progress: 100%
2021-02-12 14:16:09 (14499) [INFO]      Finalizing bias motif for scoring

2021-02-12 14:16:11 (14499) [INFO]      ----- Correcting reads from .bam within output regions -----
2021-02-12 14:16:11 (14499) [INFO]      Correction progress: 0%
2021-02-12 14:17:46 (14499) [INFO]      Correction progress: 1.0%
2021-02-12 14:17:48 (14499) [INFO]      Correction progress: 2.0%
2021-02-12 14:17:49 (14499) [INFO]      Correction progress: 3.0%
2021-02-12 14:19:22 (14499) [INFO]      Correction progress: 4.0%
2021-02-12 14:19:22 (14499) [INFO]      Correction progress: 5.0%
2021-02-12 14:19:28 (14499) [INFO]      Correction progress: 6.0%
2021-02-12 14:20:57 (14499) [INFO]      Correction progress: 7.0%
2021-02-12 14:20:58 (14499) [INFO]      Correction progress: 8.0%
2021-02-12 14:21:04 (14499) [INFO]      Correction progress: 9.0%
2021-02-12 14:22:31 (14499) [INFO]      Correction progress: 10.0%
2021-02-12 14:22:32 (14499) [INFO]      Correction progress: 11.0%
2021-02-12 14:22:40 (14499) [INFO]      Correction progress: 12.0%
2021-02-12 14:24:08 (14499) [INFO]      Correction progress: 13.0%
2021-02-12 14:24:09 (14499) [INFO]      Correction progress: 14.0%
2021-02-12 14:24:17 (14499) [INFO]      Correction progress: 15.0%
2021-02-12 14:25:44 (14499) [INFO]      Correction progress: 16.0%
2021-02-12 14:25:48 (14499) [INFO]      Correction progress: 17.0%
2021-02-12 14:25:55 (14499) [INFO]      Correction progress: 18.0%
2021-02-12 14:27:18 (14499) [INFO]      Correction progress: 19.0%
2021-02-12 14:27:23 (14499) [INFO]      Correction progress: 20.0%
2021-02-12 14:27:29 (14499) [INFO]      Correction progress: 21.0%
2021-02-12 14:28:55 (14499) [INFO]      Correction progress: 22.0%
2021-02-12 14:28:56 (14499) [INFO]      Correction progress: 23.0%
2021-02-12 14:29:07 (14499) [INFO]      Correction progress: 24.0%
2021-02-12 14:30:29 (14499) [INFO]      Correction progress: 25.0%
2021-02-12 14:30:31 (14499) [INFO]      Correction progress: 26.0%
2021-02-12 14:30:43 (14499) [INFO]      Correction progress: 27.0%
2021-02-12 14:32:05 (14499) [INFO]      Correction progress: 28.0%
2021-02-12 14:32:06 (14499) [INFO]      Correction progress: 29.0%
2021-02-12 14:32:18 (14499) [INFO]      Correction progress: 30.0%
2021-02-12 14:33:40 (14499) [INFO]      Correction progress: 31.0%
2021-02-12 14:33:42 (14499) [INFO]      Correction progress: 32.0%
2021-02-12 14:33:54 (14499) [INFO]      Correction progress: 33.0%
2021-02-12 14:35:17 (14499) [INFO]      Correction progress: 34.0%
2021-02-12 14:35:17 (14499) [INFO]      Correction progress: 35.0%
2021-02-12 14:35:35 (14499) [INFO]      Correction progress: 36.0%
2021-02-12 14:36:55 (14499) [INFO]      Correction progress: 37.0%
2021-02-12 14:36:56 (14499) [INFO]      Correction progress: 38.0%
2021-02-12 14:37:13 (14499) [INFO]      Correction progress: 39.0%
2021-02-12 14:38:33 (14499) [INFO]      Correction progress: 40.0%
2021-02-12 14:38:34 (14499) [INFO]      Correction progress: 41.0%
2021-02-12 14:38:53 (14499) [INFO]      Correction progress: 42.0%
2021-02-12 14:40:07 (14499) [INFO]      Correction progress: 43.0%
2021-02-12 14:40:08 (14499) [INFO]      Correction progress: 44.0%
2021-02-12 14:40:31 (14499) [INFO]      Correction progress: 45.0%
2021-02-12 14:41:46 (14499) [INFO]      Correction progress: 46.0%
2021-02-12 14:41:47 (14499) [INFO]      Correction progress: 47.0%
2021-02-12 14:42:12 (14499) [INFO]      Correction progress: 48.0%
2021-02-12 14:43:24 (14499) [INFO]      Correction progress: 49.0%
2021-02-12 14:43:26 (14499) [INFO]      Correction progress: 50.0%
2021-02-12 14:43:47 (14499) [INFO]      Correction progress: 51.0%
2021-02-12 14:45:02 (14499) [INFO]      Correction progress: 52.0%
2021-02-12 14:45:02 (14499) [INFO]      Correction progress: 53.0%
2021-02-12 14:45:23 (14499) [INFO]      Correction progress: 54.0%
2021-02-12 14:46:35 (14499) [INFO]      Correction progress: 55.0%
2021-02-12 14:46:37 (14499) [INFO]      Correction progress: 56.0%
2021-02-12 14:46:58 (14499) [INFO]      Correction progress: 57.0%
2021-02-12 14:48:13 (14499) [INFO]      Correction progress: 58.0%
2021-02-12 14:48:14 (14499) [INFO]      Correction progress: 59.0%
2021-02-12 14:48:33 (14499) [INFO]      Correction progress: 60.0%
2021-02-12 14:49:48 (14499) [INFO]      Correction progress: 61.0%
2021-02-12 14:49:52 (14499) [INFO]      Correction progress: 62.0%
2021-02-12 14:50:10 (14499) [INFO]      Correction progress: 63.0%
2021-02-12 14:51:25 (14499) [INFO]      Correction progress: 64.0%
2021-02-12 14:51:26 (14499) [INFO]      Correction progress: 65.0%
2021-02-12 14:51:45 (14499) [INFO]      Correction progress: 66.0%
2021-02-12 14:52:58 (14499) [INFO]      Correction progress: 67.0%
2021-02-12 14:53:01 (14499) [INFO]      Correction progress: 68.0%
2021-02-12 14:53:16 (14499) [INFO]      Correction progress: 69.0%
2021-02-12 14:54:34 (14499) [INFO]      Correction progress: 70.0%
2021-02-12 14:54:36 (14499) [INFO]      Correction progress: 71.0%
2021-02-12 14:54:49 (14499) [INFO]      Correction progress: 72.0%
2021-02-12 14:56:06 (14499) [INFO]      Correction progress: 73.0%
2021-02-12 14:56:12 (14499) [INFO]      Correction progress: 74.0%
2021-02-12 14:56:22 (14499) [INFO]      Correction progress: 75.0%
2021-02-12 14:57:39 (14499) [INFO]      Correction progress: 76.0%
2021-02-12 14:57:48 (14499) [INFO]      Correction progress: 77.0%
2021-02-12 14:57:57 (14499) [INFO]      Correction progress: 78.0%
2021-02-12 14:59:14 (14499) [INFO]      Correction progress: 79.0%
2021-02-12 14:59:24 (14499) [INFO]      Correction progress: 80.0%
2021-02-12 14:59:35 (14499) [INFO]      Correction progress: 81.0%
2021-02-12 15:00:53 (14499) [INFO]      Correction progress: 82.0%
2021-02-12 15:00:58 (14499) [INFO]      Correction progress: 83.0%
2021-02-12 15:01:07 (14499) [INFO]      Correction progress: 84.0%
2021-02-12 15:02:28 (14499) [INFO]      Correction progress: 85.0%
2021-02-12 15:02:32 (14499) [INFO]      Correction progress: 86.0%
2021-02-12 15:02:43 (14499) [INFO]      Correction progress: 87.0%
2021-02-12 15:04:04 (14499) [INFO]      Correction progress: 88.0%
2021-02-12 15:04:07 (14499) [INFO]      Correction progress: 89.0%
2021-02-12 15:04:17 (14499) [INFO]      Correction progress: 90.0%
2021-02-12 15:05:41 (14499) [INFO]      Correction progress: 91.0%
2021-02-12 15:05:44 (14499) [INFO]      Correction progress: 92.0%
2021-02-12 15:05:52 (14499) [INFO]      Correction progress: 93.0%
2021-02-12 15:07:17 (14499) [INFO]      Correction progress: 94.0%
2021-02-12 15:07:18 (14499) [INFO]      Correction progress: 95.0%
2021-02-12 15:07:28 (14499) [INFO]      Correction progress: 96.0%
2021-02-12 15:08:52 (14499) [INFO]      Correction progress: 97.0%
2021-02-12 15:08:54 (14499) [INFO]      Correction progress: 98.0%
2021-02-12 15:09:02 (14499) [INFO]      Correction progress: 99.0%
2021-02-12 15:10:21 (15631) [INFO]      Closing /mnt/home/footprint/test/myBam_uncorrected.bw (this might take some time)
2021-02-12 15:10:22 (14499) [INFO]      Correction progress: 100.0%
2021-02-12 15:10:22 (14499) [INFO]      Correction progress: done!
2021-02-12 15:10:22 (15631) [INFO]      Closing /mnt/home/footprint/test/myBam_bias.bw (this might take some time)
2021-02-12 15:10:22 (15631) [INFO]      Closing /mnt/home/footprint/test/myBam_expected.bw (this might take some time)
2021-02-12 15:10:22 (15631) [INFO]      Closing /mnt/home/footprint/test/myBam_corrected.bw (this might take some time)

2021-02-12 15:10:23 (14499) [INFO]      Verifying bias correction
2021-02-12 15:10:23 (14499) [STATS]     BIAS    pre-bias variance forward:      0.0012549
2021-02-12 15:10:23 (14499) [STATS]     BIAS    post-bias variance forward:     0.0012549
2021-02-12 15:10:24 (14499) [STATS]     BIAS    pre-bias variance reverse:      0.0012231
2021-02-12 15:10:24 (14499) [STATS]     BIAS    post-bias variance reverse:     0.0012231

2021-02-12 15:10:25 (14499) [INFO]      Finished ATACorrect run (total time elapsed: 0:55:57.233822)

```

It's clear to me that the pre and post bias correction variance is the same for forward and reverse, and I can also confirm that the file sizes in the output directory are identical:


```
-rw-rw-r-- 1 vkartha vkartha 2.2M Feb 12 14:12 myBam_AtacBias.pickle
-rw-rw-r-- 1 vkartha vkartha 6.3M Feb 12 15:06 myBam_uncorrected.bw
-rw-rw-r-- 1 vkartha vkartha 3.1K Feb 12 15:06 myBam_bias.bw
-rw-rw-r-- 1 vkartha vkartha 3.1K Feb 12 15:06 myBam_expected.bw
-rw-rw-r-- 1 vkartha vkartha 6.3M Feb 12 15:06 myBam_corrected.bw
-rw-rw-r-- 1 vkartha vkartha  31K Feb 12 15:07 myBam_atacorrect.pdf

```

Like I said the differences I'm seeing when comparing conditions make sense, along with the signal, but I wanted to make sure it wasn't incorrectly run or I'm not interpreting it wrong. Also wanted to check if it's a necessarily a problem if this happens. It does cause problems with wanting to visualize negative values too downstream with this result, since the color scales are all off

**Comment by msbentsen** [comment]
Hi, 

Thank you for posting the log - that is very helpful! This seems similar to a previous issue #57, in which there was also an issue when using "post-processed" bam-files (but which was not actually solved). I am wondering if `bedpeToBam` and/or other tools have a particular format which TOBIAS is not taking into account.... hmm.

I will have a closer look to see if I can reproduce this. Thank you for your issue report!

**Comment by vkartha** [comment]
Hi @msbentsen , thank you so much for your reply, once again, and for pointing me to the other (seemingly related) issue! That actually also made me realize I had done something almost identical to what was described in issue #57 using post-processed bam files, where I only kept the actual Tn5 cut sites (adjusted for Tn5 offset) for paired end reads, generated I converted fragment (bed) to bam format. Given TOBIAS internally does the Tn5 correction, I now realize this could lead to problems if I did use already adjusted coordinates? Is there a way to specify to skip the coordinate offsetting with TOBIAS? (of course, it is easy for me to revert to unadjusted coordinates, otherwise). Also interesting to see differences in this version compared to some earlier ones, I am using 0.12.0 here. 

Do you see any problem with only giving TOBIAS single base Tn5 cut sites (from paired ends), and the way that is processed using ATACorrect has changed over versions, or do you think it is something else?

Thanks once again for your help in advance, hope we can figure it out!

**Comment by msbentsen** [comment]
Hi @vkartha, 

I just released 0.12.9 which should fix the problem for the 1bp reads! I found a bug which falsely ignored any reads shorter than the length of the flanking region, so this is why the bias was not computed correctly. You can find the new version on Pypi and on conda later today/tomorrow. Please let me know if this solves your problem :-) @HouyuZhang in relation to #57, I believe this fix will solve your original issue as well.

If you already shifted your reads, you can set `--read_shift 0 0` within ATACorrect.

**Comment by msbentsen** [comment]
I will close this issue, but please let me know if the issues still persist.
